test/test_model.py

```diff
@@ -81,6 +86,20 @@ def test_get_relation(self):
             ['relation-list', '-r', '6', '--format=json']
         ])
 
+    def test_peer_relation_app(self):
+        meta = ops.charm.CharmMeta()
+        meta.relations = {'dbpeer': RelationMeta('peers', 'dbpeer', {'interface': 'dbpeer', 'scope': 'global'})}
+        self.model = ops.model.Model('myapp/0', meta, self.backend)
+
+        err_msg = "ERROR invalid value \"$2\" for option -r: relation not found"
+        fake_script(self, 'relation-ids',
+                    """([ "$1" = dbpeer ] && echo '["dbpeer:0"]') || echo '[]'""")
+        fake_script(self, 'relation-list',
+                    f"""([ "$2" = 0 ] && echo '[]') || (echo {err_msg} >&2 ; exit 2)""")
+
+        db1_4 = self.model.get_relation('dbpeer')
+        self.assertEqual(db1_4.app, self.model.app)
```

Comment:
I don't know if it matters, but a potentially stronger assertion here would be `assertIs`